src/Foundatio.Storage.SshNet/Storage/SshNetFileStorage.cs

```diff
@@ -13,384 +14,432 @@
+        if (searchPattern.EndsWith("/*"))
+            return await DeleteDirectory(searchPattern.Substring(0, searchPattern.Length - 2), false);
 
-            try {
-                _client.DeleteFile(NormalizePath(path));
-            } catch (SftpPathNotFoundException ex) {
-                _logger.LogDebug(ex, "Error trying to delete file: {Path}.", path);
-                return Task.FromResult(false);
-            }
+        var files = await GetFileListAsync(searchPattern, cancellationToken: cancellationToken).AnyContext();
+        int count = 0;
 
-            return Task.FromResult(true);
+        // TODO: We could batch this, but we should ensure the batch isn't thousands of files.
+        var sw = Stopwatch.StartNew();
+        _logger.LogTrace("Deleting {FileCount} files matching {SearchPattern}", files.Count, searchPattern);
+        foreach (var file in files) {
+            await DeleteFileAsync(file.Path, cancellationToken).AnyContext();
+            count++;
         }
+        _logger.LogTrace("Finished deleting {FileCount} files matching {SearchPattern} in {Duration:g}", files.Count, searchPattern, sw.Elapsed);
 
-        public async Task<int> DeleteFilesAsync(string searchPattern = null, CancellationToken cancellationToken = default) {
-            EnsureClientConnected();
-
-            if (searchPattern == null) {
-                return await DeleteDirectory("/", false);
-            } else if (searchPattern.EndsWith("/*")) {
-                return await DeleteDirectory(searchPattern.Substring(0, searchPattern.Length - 2), false);
-            }
+        return count;
+    }
 
-            var files = await GetFileListAsync(searchPattern, cancellationToken: cancellationToken).AnyContext();
-            int count = 0;
+    private void CreateDirectory(string path) {
+        string directory = Path.GetDirectoryName(path);
+        _logger.LogTrace("Ensuring {Directory} directory exists", directory);
+        
+        string[] folderSegments = directory.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
+        string currentDirectory = String.Empty;
+
+        foreach (string segment in folderSegments) {
+            currentDirectory = String.Concat(currentDirectory, "/", segment);
+            if (_client.Exists(currentDirectory)) 
+                continue;
+            
+            _logger.LogTrace("Creating {Directory} directory", directory);
+            _client.CreateDirectory(currentDirectory);
+        }
+    }
 
-            // TODO: We could batch this, but we should ensure the batch isn't thousands of files.
-            foreach (var file in files) {
-                await DeleteFileAsync(file.Path, cancellationToken).AnyContext();
+    private async Task<int> DeleteDirectory(string path, bool includeSelf) {
+        int count = 0;
+        
+        string directory = NormalizePath(path);
+        _logger.LogTrace("Deleting {Directory} directory", directory);
```

Comment:
Something like this should probably be a higher level like `info`

Reply:
Yeah, I can make this change.